### PR TITLE
[SPARK-43925][SQL][PYTHON][CONNECT] Add some, bool_or,bool_and,every to Scala, Python and Connect

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR want add some aggregate functions to Scala, Python and Connect API.
These functions show below.

- ~~any~~

- some

- every

- bool_and

- bool_or

### Why are the changes needed?
Add any, some, bool_or,bool_and,every to Scala, Python and Connect

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
New test cases.

### CODE REVIEW

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/PlanGenerationTestSuite.scala [hunk 1]
```diff
@@ -188,6 +188,7 @@ class PlanGenerationTestSuite
     .add("e", "array<int>")
     .add("f", MapType(StringType, simpleSchema))
     .add("g", "string")
+    .add("h", "boolean")
```
Comment (zhengruifeng) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/PlanGenerationTestSuite.scala:191]: let 's do not add field here, it cause too many changes.

you can add a new one if needed

**Comment by beliefer** [comment]
ping @cloud-fan @HyukjinKwon @zhengruifeng

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -1101,6 +1101,46 @@ object functions {
...
+   */
+  def bool_and(e: Column): Column = Column.fn("bool_and", e)
+
+  /**
+   * Aggregate function: returns true if at least one value of `e` is true.
+   *
+   * @group agg_funcs
+   * @since 3.5.0
+   */
+  def any(e: Column): Column = Column.fn("any", e)
```
Comment (zhengruifeng) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala:1126]: let's exclude `any` for now

File: python/pyspark/sql/functions.py [hunk 3]
```diff
@@ -2467,6 +2467,194 @@ def regr_syy(y: "ColumnOrName", x: "ColumnOrName") -> Column:
...
+    Examples
+    --------
+    >>> df = spark.createDataFrame([[True], [True], [True]], ["flag"])
+    >>> df.select(every("flag")).show()
+    +--------------+
+    |bool_and(flag)|
+    +--------------+
+    |          true|
+    +--------------+
+    <BLANKLINE>
```
Comment (zhengruifeng) [inline python/pyspark/sql/functions.py:2496]: please remove `<BLANKLINE>` in this PR

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Comment by beliefer** [comment]
@zhengruifeng Thank you!

**Pull request closed without merging**